### PR TITLE
deprecation(log): string type for `BaseHandler.formatter`

### PR DESCRIPTION
**Pull request by timreichen**
ref: https://github.com/denoland/deno_std/issues/3973#issuecomment-1869158335

### CODE REVIEW

**Comment by kt3k** [comment]
Can you also remove the example usages of string formatters in jsdoc examples?

**Comment by timreichen** [comment]
> jsdoc

done

File: log/handlers.ts [hunk 1]
```diff
@@ -16,6 +16,9 @@ export interface HandlerOptions {
 export class BaseHandler {
   level: number;
   levelName: LevelName;
+  /**
+   * @deprecated (will be removed in 0.212.0) Use {@linkcode FormatterFunction} instead of a string.
```
Comment (iuioiua) [inline log/handlers.ts:20]:
```suggestion
   * @deprecated (will be removed in 0.213.0) Use {@linkcode FormatterFunction} instead of a string.
```

**Review by iuioiua — APPROVED** [review]
Thanks!

**Review by kt3k — APPROVED** [review]
LGTM 👍

**Pull request merged**